src/transformers/modeling_reformer.py

```diff
@@ -80,7 +83,18 @@ def mish(x):
 ReformerBackwardOutput = namedtuple(
     "ReformerBackwardOutput", ["attn_output", "hidden_states", "grad_attn_output", "grad_hidden_states"]
 )
-ReformerEncoderOutput = namedtuple("ReformerEncoderOutput", ["hidden_states", "all_hidden_states", "all_attentions"])
+ReformerEncoderOutput = namedtuple(
+    "ReformerEncoderOutput", ["hidden_states", "all_hidden_states", "all_attentions", "past_buckets_states"],
+)
+
+
+def _stable_argsort(vector, dim):
+    # this function scales the vector so that torch.argsort is stable.
```

Comment:
what does stable mean here? inconsistent tiebreaking? just curious.

Reply:
yeah inconsistent tiebreaking I think it is. `e.g.` stable argmax wolud solt [1, 0, 0] always as [1, 2, 0] but unstable which is the only option for `torch.sort` can have either [1, 2, 0] or [2, 1, 0] => this function uses a trick to make sure it's always stable.